openlibrary/tests/accounts/test_models.py

```diff
@@ -1,7 +1,28 @@
-from openlibrary.accounts import model
+from openlibrary.accounts import model, InternetArchiveAccount
+from requests.models import Response
+
 
 def test_verify_hash():
       secret_key = "aqXwLJVOcV"
       hash = model.generate_hash(secret_key, "foo")
       assert model.verify_hash(secret_key, "foo", hash) == True
 
+
+class TestInternetArchiveAccount:
+      def test_xauth_error_handling_no_json(self, monkeypatch):
+            xauth = InternetArchiveAccount.xauth
+
+            resp = Response()
+            resp.status_code = 500
+            resp._content = b'Internal Server Error'
+            monkeypatch.setattr(model.requests, 'post', lambda url, **kwargs: resp)
+            assert xauth('create', s3_key='_', s3_secret='_') == {'code': 500, 'error': 'Internal Server Error'}
```

Comment:
This one does take the response code into consideration, but I think `{'code': 200, 'error': '<html>Welcome to archive.org</html>'}` is an equivalent case too. Alternatively, if the API reports its 500 or 503 or whatever in JSON, this code path is _not_ exercised. I have no idea how the external API does respond, but our code that is supposed to handle those cases doesn't tell me either way.

Reply:
I'm not testing specific codes, I'm testing http error status codes as a group; that's why I can't use 200. 400/500 are just arbitrary http errors I pulled because these are likely responses. The test is to ensure that no python error is thrown, and that the response is the expected format.

Reply:
thanks for putting that API doc link. It explains that 500s and 403s from IA will not return JSON.   All other responses do apprear to be JSON, including 503s.   The document may be implying 403 Forbidden is sometime handled with a JSON 400?